clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -118,8 +118,8 @@ public Selector(int maxReceiveSize, long connectionMaxIdleMs, Metrics metrics, T
         this.metricsPerConnection = metricsPerConnection;
     }
 
-    public Selector(long connectionMaxIdleMS, Metrics metrics, Time time, String metricGrpPrefix, Map<String, String> metricTags, ChannelBuilder channelBuilder) {
-        this(NetworkReceive.UNLIMITED, connectionMaxIdleMS, metrics, time, metricGrpPrefix, metricTags, true, channelBuilder);
+    public Selector(long connectionMaxIdleMS, Metrics metrics, Time time, String metricGrpPrefix, ChannelBuilder channelBuilder) {
+        this(NetworkReceive.UNLIMITED, connectionMaxIdleMS, metrics, time, metricGrpPrefix, metrics.config().tags(), true, channelBuilder);
```

Comment:
@junrao Sorry for late reply..

I am able to start 2 KafkaProducer and 2 KafkaConsumer with different client.id in the same jvm. Furtherore, I used JmxTool to extract all jmx metrics from this jvm, and validated that, after excluding the newly added kafka-metrics-count, jmx metric names are the same before and after my patch.

Yes that is a very good suggestion. After looking through the code, I validated that we do not need to pass tags to metricsTag in the selector constructor used by KafkaProducer/KafkaConsumer. In cases where this constructor is called, a newl instance of metrics is always created, thus excluding the possibility of sensor name conflicts.

Could you review the updated patch?